File: dev/qubit_debug.py
import os
import sys
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../qcns/python/components')))

from qubit import Qubit
from qubit import QSystem

num_qubits = 2

qsys = QSystem(num_qubits, fidelity=1, sparse=0)
qbit_0 = Qubit(qsys,0)
qbit_1 = Qubit(qsys,1)
print(f"Default system: \n{qsys}")
qbit_0.X()
print(f"Invert first Qubit: \n{qsys}")
qbit_1.X()
print(f"Invert second Qubit: \n{qsys}")
pass

File: dev/qubit_debug.cpp
#include "../cpp/components/qubit.h"
// // #include "../cpp/components/qcns_cpp/include/qcns_cpp/qubit/QSystem.h"
// // #include "../cpp/components/qcns_cpp/include/qcns_cpp/qubit/Qubit.h"
// #include "qcns_cpp/qubit/Qubit.h"
// #include "qcns_cpp/qubit/QSystem.h"
#include <chrono>

// using namespace qubit;

void 
test_old_cpp(const uint32_t& num_qubits = 1)
{
    QSystem q_sys = QSystem(num_qubits);
    Qubit qubit_1 = Qubit(&q_sys, 5);

    auto start = std::chrono::high_resolution_clock::now();
    qubit_1.X();
    auto end = std::chrono::high_resolution_clock::now();
}

void
test_new_cpp(const uint32_t& num_qubits = 1)
{
    QSystem q_sys = QSystem(num_qubits);
    Qubit qubit_1 = Qubit(&q_sys, 5);

    auto start = std::chrono::high_resolution_clock::now();
    qubit_1.X();
    auto end = std::chrono::high_resolution_clock::now();
}

int main() {
    uint32_t num_qubits = 10;
    test_old_cpp(num_qubits);
    test_new_cpp(num_qubits);

    return 0;
}